### PR TITLE
[CHI-383] Redesigned Alert component

### PR DESCRIPTION
**Pull request by mattnickles**
@SergioQCL @dani-cl-madrid

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/346/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/347/ for more details.

**Comment by mattnickles** [comment]
@jllr @janibolkvadze 
To Do:

**Tests**
- [ ] Update tests to reflect new styles and sizes 

**Custom Elements**
- [ ] Fix slot on line 92 in alert.tsx 
- [ ] Make color attribute optional now that Alerts have a base color
- [ ] Update Custom Elements Properties table to reflect new changes
- [ ] Add support for borderless attribute to banner type 
- [ ] Add support for rounded attribute to banner type
- [ ] Alert icon and alert dismiss are not rendering with their new classes m-alert__icon and m-alert__button
- [ ] Explore removing chi-icon and defining it as an icon attribute on chi-alert. This will allow us to conceal the m-alert__content div instead of requiring users to add that markup. We only need to define the icon name. Icon color is now inherited from the color attribute on chi-alert and icon size is defined via CSS based on the size of the alert.

**Examples**
- [ ] Ensure all Alerts used throughout Chi and Chi Projects are rendering the new alert

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/348/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/363/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/367/ for more details.

**Comment by jllr** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/368/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/371/ for more details.

**Comment by jllr** [comment]
@mattnickles please solve conflicts and so we can merge it

File: cypress/integration/chi-custom-elements/alert.spec.js [hunk 1]
```diff
@@ -98,11 +117,11 @@ describe('Alert', () => {
     it('Alert banner dismiss button should trigger appropriate event', () => {
       const spy = cy.spy();
 
-      cy.get('body').then((el) => {
+      cy.get('body').then(el => {
         el.on('dismissAlert', spy);
       });
 
-      cy.get('@alert-banner-large-success-dismiss')
+      cy.get('@alert-banner-dismissible-success-default')
```
Comment (SergioQCL) [inline cypress/integration/chi-custom-elements/alert.spec.js]: Sizes do not match here (replaced and new). Is this correct?

Comment (janibolkvadze) [reply cypress/integration/chi-custom-elements/alert.spec.js]: Yes, it's correct. We added several tests / changed existing ones. That particular test checks if clicking chi-button inside of the alert triggers the appropriate event, size is irrelevant in this case.

File: src/website/views/components/alert.pug [hunk 2]
```diff
@@ -6,253 +6,998 @@ p.-text Alert messages provide contextual feedback for user actions.
...
+      </div>
+      <button class="m-alert__dismiss-button a-btn -icon -close">
+        <div class="a-btn__content">
+          <i class="a-icon icon-x"></i>
         </div>
+      </button>
+    </div>
+
+
+
```
Comment (SergioQCL) [inline src/website/views/components/alert.pug]: Please, remove unnecessary blank lines

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/382/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/383/ for more details.

File: src/custom-elements/src/components/alert/alert.tsx [hunk 3]
```diff
@@ -23,12 +23,27 @@ export class Alert {
...
-   *  to set size of an alert, lg is the only size available.
+   *  to avoid necessity of adding <chi-icon> to alert markup.
    */
-  @Prop({ reflectToAttr: true }) size = 'md';
+  @Prop({ reflectToAttr: true }) icon: string;
+
+  /**
+   *  to set alert size { sm, md, lg }.
+   */
+  @Prop({ reflectToAttr: true }) size = '';
```
[the remark this answers is not in the document]
Comment (mattnickles) [reply src/custom-elements/src/components/alert/alert.tsx]: 👍 set md as default

File: src/website/views/components/alert.pug (hunk 2 above)
Comment (mattnickles) [reply src/website/views/components/alert.pug]: blank lines have been removed

**Comment by mattnickles** [comment]
Items pending:

Custom Elements

- [ ] Make m-alert__text invisible
- [ ] Change title attribute from `alert-title` to `title`
- [ ] Make m-alert__actions invisible

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/389/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/390/ for more details.

**Pull request merged**